File: java/lib/src/main/java/io/openweb3/walletpay/Transfers.java
package io.openweb3.walletpay;

import io.openweb3.walletpay.exceptions.ApiException;
import io.openweb3.walletpay.internal.api.TransfersApi;
import io.openweb3.walletpay.models.TransferRequest;
import io.openweb3.walletpay.models.TransferResponse;

public final class Transfers {
	private final TransfersApi api;

	Transfers() {
		api = new TransfersApi();
	}

	public TransferResponse create(final TransferRequest transferIn) throws ApiException {
		return this.create(transferIn, new PostOptions());
	}

	public TransferResponse create(final TransferRequest transferIn, final PostOptions options) throws ApiException {
		try {
			return api.v1TransfersTransfer(transferIn);
		} catch (io.openweb3.walletpay.internal.ApiException e) {
			throw Utils.WrapInternalApiException(e);
		}
	}
}

File: go/transfers.go
package walletpay

import (
	"context"

	"github.com/openweb3-io/wallet-pay-openapi/go/internal/openapi"
)

type (
	TransferRequest  = openapi.TransferRequest
	TransferResponse = openapi.TransferResponse
)

type Transfers struct {
	api *openapi.APIClient
}

func (e *Transfers) Create(ctx context.Context, transferIn *TransferRequest) (*TransferResponse, error) {
	req := e.api.TransfersApi.V1TransfersTransfer(ctx)
	req = req.TransferRequest(openapi.TransferRequest(*transferIn))

	out, res, err := req.Execute()
	if err != nil {
		return nil, wrapError(err, res)
	}
	ret := TransferResponse(out)

	return &ret, nil
}
